common/journal.cpp

```diff
@@ -151,17 +150,17 @@ int reinitializeJournal() {
 	std::strftime(dateStr, FILENAME_DATESTR_MAX_LENGTH, FILENAME_DATESTR_FORMAT, &now_tm);
 
 	// Form base filename from path and label
-	UtilGetJournalDirectoryPath(journalDirPath, sizeof (journalDirPath));
+	auto journalDirPath = Util::getDirectoryPath("journal");
 	journalPath = journalDirPath + journalLabel + "-" + dateStr + JOURNAL_FILE_ENDING;
 
 	// Check if log directory exists
-	if(stat(journalDirPath, &sb)) {
+	if(stat(journalDirPath.c_str(), &sb)) {
 		// Create directory if not
-		if (mkdir(journalDirPath, 0775) != -1) {
-			RCLCPP_INFO(rclcpp::get_logger(logName), "Created directory %s", journalDirPath);
+		if (mkdir(journalDirPath.c_str(), 0775) != -1) {
+			RCLCPP_INFO(rclcpp::get_logger(logName), "Created directory %s", journalDirPath.c_str());
 		}
 		else {
-			RCLCPP_ERROR(rclcpp::get_logger(logName), "Unable to create directory <%s>", journalDirPath);
+			RCLCPP_ERROR(rclcpp::get_logger(logName), "Unable to create directory <%s>", journalDirPath.c_str());
 			return -1;
 		}
 	}
```

Comment:
I agree, but I think this should be done in another PR. This PR was mainly to rewrite functions from `util.c`, otherwise this PR would be very large if all of this code also gets rewritten.